### PR TITLE
[ren] Separate INT/FP Rename Instantiations

### PR DESCRIPTION
**Pull request by bkorpan**
**Type of change**: other enhancement

**Impact**: rtl refactoring

**Development Phase**: implementation

**Release Notes**
Describe the INT/FP pipelines as separate instantiations of a module, rather than a single instantiation.

### CODE REVIEW

File: src/main/scala/exu/core.scala [hunk 1]
```diff
@@ -438,26 +443,52 @@ class BoomCore(implicit p: Parameters, edge: freechips.rocketchip.tilelink.TLEdg
...
+  }
 
+  // Outputs
   dis_uops := rename_stage.io.ren2_uops
   dis_valids := rename_stage.io.ren2_mask
 
+  /**
+   * TODO This is a bit nasty, but it's currently necessary to
+   * split the INT/FP rename pipelines into separate instantiations.
+   * Won't have to do this anymore with a properly decoupled FP pipeline.
```
Comment (jerryz123) [inline src/main/scala/exu/core.scala:507]: How do you plan on handling int-to-fp/fp-to-int moves without doing this ugly merge step?

Comment (bkorpan) [reply src/main/scala/exu/core.scala:507]: If the FP rename pipeline ends up living behind a queue, fp-to-int could be handled the same as it is now. int-to-fp would need to use some sort of structure to match up fp destinations to the uops coming from the int side.

**Comment by bkorpan** [comment]
This boots fedora.

File: src/main/scala/exu/core.scala [hunk 2]
```diff
@@ -104,7 +104,11 @@ class BoomCore(implicit p: Parameters, edge: freechips.rocketchip.tilelink.TLEdg
 
   val decode_units     = for (w <- 0 until decodeWidth) yield { val d = Module(new DecodeUnit); d }
   val dec_brmask_logic = Module(new BranchMaskGenerationLogic(coreWidth))
-  val rename_stage     = Module(new RenameStage(coreWidth, numIntRenameWakeupPorts, numFpWakeupPorts))
+  val rename_stage     = Module(new RenameStage(coreWidth, numIntPhysRegs, numIntRenameWakeupPorts, false))
```
Comment (jerryz123) [inline src/main/scala/exu/core.scala:107]: Nit, but naming this `int_rename_stage` might make more sense,

Although if FP-pipeline becomes more decoupled, this naming is fine,

**Review by jerryz123 — APPROVED** [review]

**Comment by bkorpan** [comment]
This needs rebasing for a very simple merge conflict.

**Pull request merged**